Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/CallbackJoiner.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+import StitchCore
+import XCTest
+import MongoSwift
+@testable import StitchCoreRemoteMongoDBService
+
+private class SynchronizedDispatchDeque {
+    private let rwLock = ReadWriteLock()
+    private var workItems = [DispatchWorkItem]()
+    var count: Int {
+        return workItems.count
+    }
+    
+    func append(_ workItem: DispatchWorkItem) {
+        rwLock.writeLock()
+        defer { rwLock.unlock() }
+        workItems.append(workItem)
+    }
+
+    func removeFirst() -> DispatchWorkItem {
+        rwLock.writeLock()
+        defer { rwLock.unlock() }
+        return workItems.removeFirst()
+    }
+}
+
+/// Synchronously capture the value of a asynchronous callback.
+class CallbackJoiner {
```

Comment:
[nit] `CallbackJoiner` could maybe be called `CallbackAwaiter`, since we use `await` in the other SDKs, but I'm also fine keeping it as is

Reply:
fair enough, I'm happy with `CallbackJoiner`